cfgov/v1/models/learn_page.py

```diff
@@ -60,16 +60,23 @@ class AbstractFilterPage(CFGOVPage):
         MultiFieldPanel(Page.settings_panels, 'Scheduled Publishing'),
     ]
 
-    edit_handler = TabbedInterface([
-        ObjectList(CFGOVPage.sidefoot_panels, heading='Sidebar'),
-        ObjectList(settings_panels, heading='Configuration'),
-    ])
-
     # This page class cannot be created.
     is_creatable = False
 
     objects = CFGOVPageManager()
 
+    @classmethod
+    def generate_edit_handler(self, content_panel):
+        content_panels = [
+            StreamFieldPanel('header'),
+            content_panel,
+        ]
+        return TabbedInterface([
+            ObjectList(self.content_panels + content_panels, heading='General Content'),
+            ObjectList(CFGOVPage.sidefoot_panels, heading='Sidebar'),
+            ObjectList(self.settings_panels, heading='Configuration'),
+        ])
+
```

Comment:
Style note: it's pretty standard I think to use 'cls' as the first argument to a class method.

What I was getting at in chat, was the idea that edit_handler could become a method, and sniff out whether or not it's appropriate to include StreamField('content') based on examining the value of content_panels, or whether the class has a member called 'content'. That would eliminate the boilerplate you've had to add to the subclasses.

I now think there's a better idea, though: we could override the [get_edit_handler](https://github.com/torchbox/wagtail/blob/db1e4aad1585a66451ae6e7de8f0af7ad84df054/wagtail/wagtailadmin/edit_handlers.py) method that Wagtail pages have to do that same thing.

it could look like:

```python
@classmethod
def get_edit_handler(cls):
    if hasattr(cls,'content'):
        cls.content_panels += StreamFieldPanel('content')
    return super(AbstractFilterPage, cls).get_edit_handler()
```